src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1684,22 +1682,14 @@ protected override void ProcessRecord()
                             || (ServiceStartMode)(-1) != StartupType)
                         {
                             DWORD dwStartType = NativeMethods.SERVICE_NO_CHANGE;
-                            switch (StartupType)
+                            if ((ServiceStartMode)(-1) != StartupType &&
```

Comment:
It seems `(ServiceStartMode)(-1) != StartupType &&` should be removed.

Reply:
Will change, but will need to change some other code to make this work.  Seems like `-1` shouldn't be needed at all.

Reply:
Actually, still need `-1` which will be treated as equivalent to `SERVICE_NO_CHANGE`

Reply:
By looking at the original code, for `New-Service`, `SERVICE_AUTO_START` would be used in case `StartupType` is -1. So it seems we cannot just use `dwStartType = NativeMethods.SERVICE_NO_CHANGE` for `(ServiceStartMode)(-1)`

Reply:
New-Service [defaults](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs#L1933) to Automatic whereas Set-Service [defaults](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs#L1511) to -1 to indicate no change.

Reply:
Please correct me if I'm wrong. The following snippet is the original code in `New-Service`. It indicates `dwStartType` should be `SERVICE_AUTO_START` when `StartupType == ((ServiceStartMode)(-1))`
```
DWORD dwStartType = NativeMethods.SERVICE_AUTO_START;
switch (StartupType)
{
    case ServiceStartMode.Automatic:
        dwStartType = NativeMethods.SERVICE_AUTO_START;
        break;
    case ServiceStartMode.Manual:
        dwStartType = NativeMethods.SERVICE_DEMAND_START;
        break;
    case ServiceStartMode.Disabled:
        dwStartType = NativeMethods.SERVICE_DISABLED;
        break;
    default:
        Diagnostics.Assert(
            ((ServiceStartMode)(-1)) == StartupType,
            "bad StartupType");
        break;
}
```
It's replaced with `if (!NativeMethods.TryGetNativeStartupType(StartupType, out DWORD dwStartType)) { // write error }` after the change, and `TryGetNativeStartupType` will set `dwStartType` to `SERVICE_NO_CHANGE` when `StartupType == ((ServiceStartMode)(-1))`.

So it looks to me changes the value of `dwStartType` in case of `StartupType == ((ServiceStartMode)(-1))`.

Reply:
I'll fix it so if it's (-1), it should default to Auto for New-Service